WalletWasabi/Wallets/IRepository.cs

```diff
@@ -8,7 +8,7 @@ namespace WalletWasabi.Wallets
 	/// </summary>
 	public interface IRepository<TID, TElement>
 	{
-		Task<TElement> GetAsync(TID id, CancellationToken cancel);
+		Task<TElement?> GetAsync(TID id, CancellationToken cancel);
```

Comment:
Can you rename it to `TryGetAsync`? `GetAsync` suggest that if not found, then it'd throw an exception and it never returns null, but if you call it `TryXXX` then it suggests it'll return an out parameter or failing that because async functions cannot do that, it'll return null if not found.